src/mocsolver.jl

```diff
@@ -205,8 +254,44 @@ function compute_q!(sol::MoCSolution{T}, prob::MoCProblem) where {T}
             #     1 / keff * χ[g] * νΣf[g′] * φ[@region_index(i, g′)]
             #     for g′ in 1:NGroups
             # )
+            qig /= 4π
+            q[ig] = qig
+        end
+    end
 
-            qig /= (4π * Σt[g])
+    return nothing
+end
+
+function compute_q!(sol::MoCSolution{T}, prob::MoCProblem, fixed_sources::Matrix{Float64}) where {T}
+    NGroups = ngroups(prob)
+    NRegions = nregions(prob)
+    @unpack keff, φ, q = sol
+
+    @inbounds for i in 1:NRegions
+        xs = getxs(prob, i)
+        @unpack χ, Σt, νΣf, Σs0 = xs
+        fissionable = isfissionable(xs)
+
+        for g in 1:NGroups
+            ig = @region_index(i, g)
+            qig = zero(T)
+            for g′ in 1:NGroups
+                ig′ = @region_index(i, g′)
+                qig += Σs0[g′, g] * φ[ig′]
+                if fissionable
+                    qig += 1 / keff * χ[g] * νΣf[g′] * φ[ig′]
+                end
+            end
+            # qig = sum(
+            #     Σs0[g′, g] * φ[@region_index(i, g′)] +
+            #     1 / keff * χ[g] * νΣf[g′] * φ[@region_index(i, g′)]
+            #     for g′ in 1:NGroups
+            # )
+            qig += fixed_sources[i,g]
+            qig /= 4π 
+            if qig < 0.0
+                qig = 1e-10
+            end
```

Comment:
How could this be negative you never subtracted to `qig`?

Reply:
I based this off of [these lines](https://github.com/mit-crpg/OpenMOC/blob/7c8c9460c1c95f68dae102a402a39afa233a0b8c/src/CPUSolver.cpp#LL1981C1-L1982C48) in OpenMOC, but maybe they aren't necessary